examples/psyir/newnode.py

```diff
@@ -0,0 +1,83 @@
+# -----------------------------------------------------------------------------
```

Comment:
I moved it to `doc/developer_guide/code_snippets` and sphinx imports it, but it is not being executed (tested) and won't warn if its out-of-date (could be done in another issue?). Still I think it will potentially make the documentation easier to update in the future if relevant interfaces change.